src/main/kotlin/com/zepben/auth/client/AuthProviderConfig.kt

```diff
@@ -0,0 +1,140 @@
+    val response = client.send(HttpRequest.newBuilder().uri(URI(issuerURL)).GET().build(), handler)
+    if (response.statusCode() == StatusCode.OK.code) {
+        try {
+            val authConfigJson = Json.decodeValue(response.body()) as JsonObject
+            return ProviderDetails(
+                tokenEndpoint = authConfigJson.getString("token_endpoint", ""),
+                jwkUrl = authConfigJson.getString("jwks_uri", ""),
+                issClaim = authConfigJson.getString("issuer", "")
+            )
+
+        } catch (e: DecodeException) {
+            throw AuthException(
+                response.statusCode(),
+                "Expected JSON response from $issuerURL, but got: ${response.body()}."
+            )
+        } catch (e: ClassCastException) {
+            throw AuthException(
+                response.statusCode(),
+                "Expected JSON object from $issuerURL, but got: ${response.body()}."
+            )
+        }
+    } else {
+        throw AuthException(
+            response.statusCode(),
+            "$issuerURL responded with error: ${response.statusCode()} - ${response.body()}"
+        )
+    }
+}
+
+// Fetch info from EWB first (issuer, audience) using the confAddress, and then fetch the rest from the provider
+fun createProviderConfig(
+    confAddress: String,
+    client: HttpClient = HttpClient.newBuilder().sslContext(SSLContextUtils.allTrustingSSLContext()).build(),
+    handler: HttpResponse.BodyHandler<String> = HttpResponse.BodyHandlers.ofString(),
+    authTypeField: String = "authType",
+    audienceField: String = "audience",
+    protocolField: String = "protocol",
+    issuerField: String = "issuer",
+    fetchProviderDetails: Boolean = true
+): AuthProviderConfig {
+
+    // Fetch the auth data from EWB
+    var ewbUrl: String
+    val response = try {
+        // Try with https first
+        client.send(HttpRequest.newBuilder().uri(URI(buildEwbUrl("https", confAddress))).GET().build(), handler)
+    } catch (e: IOException) {
+        // Try with http
+        client.send(HttpRequest.newBuilder().uri(URI(buildEwbUrl("http", confAddress))).GET().build(), handler)
+    }
+
+    return response?.let {
+        if (response.statusCode() == StatusCode.OK.code) {
+            try {
+                val authConfigJson = Json.decodeValue(response.body()) as JsonObject
+                val issuer = authConfigJson.getString(issuerField, "")
+                val authMethod = AuthMethod.valueOf(authConfigJson.getString(authTypeField, "none").uppercase())
+
+                if (authMethod == AuthMethod.NONE) {
+                    throw AuthException(1, "Detected Auth set to NONE, this is not supported for fetching tokens! Check your configuration matches $confAddress")
+                }
+
+                val audience = authConfigJson.getString(audienceField, "")
+                val protocol = authConfigJson.getString(protocolField, "https")
```

Comment:
Question, instead of making this throw an exception, would it be better to have upstream just return a zepben token fetcher that did a noop fetch? Or is it better to isolate oauth vs none auth?